cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -853,15 +858,44 @@ Future<ReconciliationState> reconcileEntityOperator(Clock clock)    {
         }
 
         /**
-         * @return true if the autorebalance is enabled within the Cruise Control configuration, false otherwise
+         * Based on the autorebalance being enabled or not within Cruise Control configuration:
+         * - return true because the auto-rebalancing reconciliation has to run or ...
+         * - ... create the auto-rebalance status on Idle if the cluster is still in the creation phase (no auto-rebalancing to run yet) or ...
+         * - ... reset to null the auto-rebalance status because the auto-rebalance was disabled
+         *
+         * @return true if the autorebalance reconciliation has to run, false otherwise
          */
-        boolean isAutoRebalancingEnabled() {
-            // Cruise Control is not defined in the Kafka custom resource or no autorebalance enabled, so nothing to reconcile
+        boolean maybeAutoRebalancing() {
```

Comment:
This suggests the method might do auto-rebalancing. So I think this is confusing and we should try to find a better name. I also wonder if with the added logic it would be better to hide it into `reconcileKafkaAutoRebalancing` instead of using it as a condition on the step on line 288. (i.e. call `reconcileKafkaAutoRebalancing` everytime and make the decision inside)

Reply:
I am fine with putting the code into `reconcileKafkaAutoRebalancing` because I cannot think a better name for `maybeAutoRebalancing`.